buildspecs/al-maintenance-release-config.yml

```diff
@@ -0,0 +1,47 @@
+version: 0.2
+
+phases:
+  build:
+    commands:
+      - GITHUB_DEST_BRANCH_NAME=$(echo ${CODEBUILD_WEBHOOK_BASE_REF} | cut -d "/" -f3-)
+      - GIT_COMMIT_SHA=$(git rev-parse $GITHUB_DEST_BRANCH_NAME)
+      - GIT_COMMIT_SHORT_SHA=$(git rev-parse --short=8 $GITHUB_DEST_BRANCH_NAME)
+      - RELEASE_DATE=$(date +'%Y%m%d')
+      - AGENT_VERSION=$(cat VERSION)
+
+      - |
+        cat << EOF > agent.json
+        {
+          "agentReleaseVersion" : "$AGENT_VERSION",
+          "releaseDate" : "$RELEASE_DATE",
+          "initStagingConfig": {
+            "release": "1"
+          },
+          "agentStagingConfig": {
+            "releaseGitSha": "$GIT_COMMIT_SHA",
+            "releaseGitShortSha": "$GIT_COMMIT_SHORT_SHA",
+            "gitFarmRepoName": "MadisonContainerAgentExternal",
+            "gitHubRepoName": "aws/amazon-ecs-agent",
+            "gitFarmStageBranch": "v${AGENT_VERSION}-stage",
+            "githubReleaseUrl": ""
+          }
+        }
+        EOF
+
+      # Downloading the agentVersionV2-<branch>.json (it is assumed that it already exists)
+      - aws s3 cp ${RESULTS_BUCKET_URI}/agentVersionV2/agentVersionV2-${GITHUB_SOURCE_BRANCH_NAME}.json ./agentVersionV2-${GITHUB_SOURCE_BRANCH_NAME}.json
```

Comment:
Something that I've just realized that we haven't handled in any of the other scripts either is updating these binaries to the latest version (`aws` and `jq`). The potential problem we could run into here is if a new feature gets introduced in the AWS CLI and we haven't picked the latest version but use the feature. Edge case for sure, but what do you think about including a `yum update` as an `install` or `pre_build` step?

Reply:
Yeah that's a really good point. I'll add that.

Reply:
Added a pre_build stage that'll run yum update.